### PR TITLE
Language Server Protocol (LSP) Support

### PR DESCRIPTION
**Pull request by drkstr101**
Xtext has undergone a large upgrade in order to provide IDE support
	outside of Eclipse. This could, for example, enable VSCode or even
	pure web based editors. This update is the first of many (TBD)
	related feature updates to fully leverage the power and flexibility
	of the LSP within Natural. Updates Include:

	1. New Tycho builder for easier CI and release builds
	2. Improved developer test workflow
	3. Includes the new XtextGenerator that generates the LSP components
	4. Upgrades CucumberFormatter to the new formatting2 API, allowing
	   for more fine-grained control of formatting rules
	5. Automated step matching now runs in background, and has been
	   updated to match Cucumber JVM5 (TODO: this could cause regression
	   due to the default changing. Should possibly be changed to
	   require explicit action from the user, such as a setting change.)

### CODE REVIEW

File: org.agileware.natural.cucumber.feature/build.properties [hunk 1]
```diff
@@ -1,7 +1,5 @@
-bin.includes = feature.xml,\
-               feature.properties,\
-               license.html
-src.includes = license.html,\
-               feature.xml,\
-               feature.properties,\
-               build.properties
+bin.includes = feature.xml,\
```
Comment (drkstr101) [inline org.agileware.natural.cucumber.feature/build.properties:1]: Missing license.html

**Comment by drkstr101** [comment]
@rlogiacco This PR is now ready for **release candidate** testing. I was going to try and improve the formatter a bit, but how to implement this is somewhat opinionated, so it  prob makes sense to put this in a separate feature branch.

**Comment by drkstr101** [comment]
@rlogiacco Version numbers have been reset as per our discussion. Please let me know if this is not what you had in mind.

Cheers!

**Comment by rlogiacco** [comment]
I'm not sure I understand the version numbering...
I know this mess is originated by me and I'm willing to find a solution
together, it doesn't have to be mine: I'm really open to proposals and
alternative paths.

Our latest release was cucumber-0.7.7 and jbehave-0.3.1
<https://github.com/rlogiacco/Natural/commit/278900731425a19849ac2bbfb289353b031cf3d2>,
both being part of natural-1.3.6.

I was reserving version 1.0.0 (Cucumber?) for the first stable release
covering all language features, but this has not to remain like that.

Let's imagine we go for releasing only the gherkin part, how do we call it?
Natural for Cucumber? If that's the case, which version are we releasing?
Are we kicking in with 1.0.0? Or do we consider it 0.8.0?

Are we also releasing Natural for JBehave? And which version is that? I
believe we are too far from completion to call it 1.0.0... is it 0.4.0?

Can we drop once and for all the version of Natural and just version the
core as we find fit without having to use it on comms?

A lot of questions, I know... and all that is just confusing... sorry :-)

**Comment by drkstr101** [comment]
I think I should probably defer to you on this one due to my lack of experience with the subject matter. I have never released an Eclipse plugin before, so I'm not really sure which concerns are important or not.

As far as I can tell, there is no practical reason for there to be a "Natural" version number. This only really exists because I needed something to enter into the parent pom.xml. Or maybe I don't? I'm not to sure how/when it relates.

I think my main confusion is on the disconnect between the version numbers in `pom.xml`, `feature.xml`, and `MANIFEST.mf`. It may be good idea to get these all (common, cucumber, jbehave) sync somehow. I mean to say that they would all have their own version, but be consistent across those 3 files in that bundle.

Also FYI I have what is essentially `jbehave2` in beta. Its has a few rough edgest, but a lot of bug fixes and better support for official spec. It was basically my "learning exercise" for work that went into Cucumber. I wasn't sure if you wanted to bring in the extra changes as this PR is pretty big as is.

**Comment by rlogiacco** [comment]
I suggest to hold on jbehave.

Keeping the three versions aligned is not impossible, just requires some
work and it might impact the eclipse project, which is the reason I didn't
wasn't to mess with that right now

**Comment by drkstr101** [comment]
Just to clarify, I didn't mean to say common, cucumber, & jbehave modules should all have the same versions, but that pom.xml, feature.xml, & MANIFEST.mf should all have the same versions across these files within the same module.

I believe we can just set these numbers to whatever you like. I don't know if a version number is required or not in parent pom.xml, but it doesn't seem to have any affect as the build output is the same regardless. If it does require one to be set, we can just keep it at `1.0.0-SNAPSHOT` forever, I would assume.

So my understanding is there are three separate "modules" which are versioned seperatly. These may all be updated and released separately, although `cucumber` and `jbehave` must keep API compatibility with `common` as it is a direct dependency.

If we are all on the same page with the above, than I think all we need to do is come up with those 3 numbers for this release. Just let me know what you want those to be, or if I may be able to answer any questions on what's changed. Come to think of it, maybe I should just reset `jbehave` back to master, and `common ` could probably get away with just a minor bump. 

I am not quite sure this is `1.0.0` ready just yet though. My guess is maybe like 2 or 3 release candidates away. I was really hoping to have fully ANTLR3 compliant grammar by `1.0.0` release. Most of ths work is already done, I just didn't want to overload this PR with to much from `beta` at once, and have to test a million unrelated minor issues with all the changes.

**Pull request merged**